src/core/pipeline/pipeline.js

```diff
@@ -0,0 +1,35 @@
+export default class Pipeline {
```

Comment:
Here's brief list of what needs to be done. Guys, please have a look at it
1. Pipeline's stage exception handling
2. Timeouts?
3. Debounce?